### DISCUSSION TITLE
How to hide a `coder_app` resource?

### DISCUSSION THREAD

**Discussion by ElioDiNino** [coder/coder#11349] [General] [answered]
I want to build a template where a user can choose whether or not they want `coder-server` installed and available. I have this working with a boolean parameter and the following within the startup script:

```bash
if "${data.coder_parameter.code_server.value}"
then
  # install and start code-server
  curl -fsSL https://code-server.dev/install.sh | sh -s -- --method=standalone --prefix=/tmp/code-server
  /tmp/code-server/bin/code-server --auth none --port 13337 >/tmp/code-server.log 2>&1 &
fi
```

The issue is that there is no way to render the associated `coder_app` resource conditionally, so if a user chooses not to install coder-server through the parameter then they get the following instead of it being hidden.

![Screenshot 2023-07-05 at 10 50 31 AM](https://github.com/coder/coder/assets/52972357/28c61a0f-6d1a-4e1c-9284-a888b62236de)

Adding a boolean `hide` field like `coder_metadata` has would allow template creators to conditionally render any `coder_app` they want which would enable users choosing which apps they want available through parameters without confusing UI results (i.e. my screenshot above)

**Comment by kylecarbs** [comment]
Good feature! Thanks :)

**Comment by MrPeacockNLB** [comment, marked as answer]
We use **count** in Terraform to enable/disable a coder_app resource. But we know at provision time if a coder_app is enabled or not.

This should work in your case as you already know if the developer wants a coder instance.

```
resource "coder_app" "code-server" {
  count        = (data.coder_parameter.code_server.value == true) ? 1 : 0

  agent_id     = coder_agent.main.id
  slug         = "code-server"
  display_name = "code-server"
...
```

**Comment by ElioDiNino** [comment]
> We use **count** in Terraform to enable/disable a coder_app resource. But we know at provision time if a coder_app is enabled or not.
> 
> This should work in your case as you already know if the developer wants a coder instance.
> 
> ```
> resource "coder_app" "code-server" {
>   count        = (data.coder_parameter.code_server.value == true) ? 1 : 0
> 
>   agent_id     = coder_agent.main.id
>   slug         = "code-server"
>   display_name = "code-server"
> ...
> ```

Oh that's perfect! I should have tried that first thanks. Everything works as expected with this solution so I will close this issue

**Comment by sharkymark** [comment]
Cool stuff here. I made a [template with true false to install Coder's code-server](https://github.com/sharkymark/v2-templates/tree/main/pod-maybe-code-server)

And another [template if Microsoft or Coder's code-server is needed](https://github.com/sharkymark/v2-templates/tree/main/pod-optional-code-server)
